---
title: "R1_FHAB_seq_edit"
author: "K_Langlois"
date: "`r Sys.Date()`"
output: pdf_document
---

```{r setup, include = T, echo=F}
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
#not include "echo" or "eval" --> do individually to chunks
```


## Code to be used for R1 & statewide FHAB sequencing data

### set up
install packages

```{r install, echo = T}
library(knitr)
library(seqinr)
library(reshape2)
library(ggplot2)
library(vegan)
library(stringr)
library(plyr)
library(dplyr)
library(ggtree)
set.seed(100)
```

upload files

```{r file_16S, echo = T}
path<-"/Users/kylielanglois/OneDrive - SCCWRP/algae/FHABs/"
d<-read.csv(file.path(path, "R1_16Scyano_022420_75_table.csv"))

met<-read.csv(file.path(path, "R1_statewide_FHAB_metadata_update_tax.csv"))

tax.95<-read.delim(
  file.path(path, "R1_16Scyano_022420_75_CyanoSeq_update95_bl_tax.tsv"))
tax.97<-read.delim(
  file.path(path, "R1_16Scyano_022420_75_CyanoSeq_update97_bl_tax.tsv"))

tox<-read.csv(file.path(path, "Cyanotoxin_production_summ_AL.csv"))

```

```{r file_options, echo = T}
d.asv<-d
samps.sum<-data.frame(sample=colnames(d[, 2:ncol(d)]), 
                      sum=colSums(d[, 2:ncol(d)]))
samps.sum<-merge(samps.sum,
                 met[, match(c("Sequencing.name.16Scyano", "Station.Name"), 
                 #met[, match(c("Sequencing.name.anaC", "Station.Name"), 
                             colnames(met))], 
                 by.x="sample", by.y="Sequencing.name.16Scyano", all.x=T)
samps.sum.2<-plyr::ddply(samps.sum, "Station.Name", plyr::numcolwise(sum))
samps.sum.2<-samps.sum.2[!grepl("EB", samps.sum.2$Station.Name), ]
met1<-met
met1[is.na(met1)]<-"0"

```

```{r file_edit, echo=T, eval=T}
#combine----
tax.combo<-merge(tax.95, tax.97, by="Feature.ID")
colnames(tax.combo)<-gsub("\\.x", "_95", colnames(tax.combo))
colnames(tax.combo)<-gsub("\\.y", "_97", colnames(tax.combo))
tax.combo$agree<-tax.combo$Taxon_95==tax.combo$Taxon_97 #which agrees fully

#if no ID from 97%, use ID from 95%
tax.combo$combo_tax<-ifelse(tax.combo$agree==TRUE, tax.combo$Taxon_97, 
                            ifelse(tax.combo$Taxon_97=="Unassigned", 
                                   tax.combo$Taxon_95, tax.combo$Taxon_95))
tax.combo.1<-tax.combo[, match(c("Feature.ID", "combo_tax"), 
                               colnames(tax.combo))]
tax.combo.2 <- as.data.frame(cbind(
  tax.combo.1$Feature.ID,str_split_fixed(
    tax.combo.1$combo_tax, ";", 7))) #split into columns
colnames(tax.combo.2)<-c("ASV", "taxon1", "taxon2", "taxon3", "taxon4", 
                         "taxon5", "taxon6", "taxon7") #rename new columns
tax.combo.2 <- tax.combo.2 %>% mutate_all(na_if,"")
tax.combo.2$taxon1<-ifelse(is.na(tax.combo.2$taxon1), NA, 
                           paste("t1", tax.combo.2$taxon1, sep="_"))
tax.combo.2$taxon2<-ifelse(is.na(tax.combo.2$taxon2), NA, 
                           paste("t2", tax.combo.2$taxon2, sep="_"))
tax.combo.2$taxon3<-ifelse(is.na(tax.combo.2$taxon3), NA, 
                           paste("t3", tax.combo.2$taxon3, sep="_"))
tax.combo.2$taxon4<-ifelse(is.na(tax.combo.2$taxon4), NA, 
                           paste("t4", tax.combo.2$taxon4, sep="_"))
tax.combo.2$taxon5<-ifelse(is.na(tax.combo.2$taxon5), NA, 
                           paste("t5", tax.combo.2$taxon5, sep="_"))
tax.combo.2$taxon6<-ifelse(is.na(tax.combo.2$taxon6), NA, 
                           paste("t6", tax.combo.2$taxon6, sep="_"))
tax.combo.2$taxon7<-ifelse(is.na(tax.combo.2$taxon7), NA, 
                           paste("t6", tax.combo.2$taxon7, sep="_"))
tax.combo.2$last_tax<-apply(tax.combo.2, 1, function(x) tail(na.omit(x), 1))

tax.select<-tax.combo.2[, match(c("ASV", "taxon2", "taxon4", "last_tax"), colnames(tax.combo.2))]
colnames(tax.select)[4]<-"taxon6" #to make code work
#gen<-tax.ana[, match(c("Feature.ID", "Taxon4", "Taxon6"), colnames(tax.ana))] #anaC
#just genus assignments

#combine with data table----
d.asv.gen<-merge(d.asv, tax.select, by="ASV", all.x=T, sort=F)
d.asv.gen$taxon6<-gsub("t6_", "", d.asv.gen$taxon6)
#d.asv.gen<-merge( gen, d.asv, by.y="ASV", by.x="Feature.ID", all.y=T, sort=F) #anaC
```

toxin info 
```{r tox_1, echo=T, eval=T}
#edit toxin info--------
tox1<-as.data.frame(apply(tox, 2, function(x) gsub("^$|^ $", NA, x))) 
#change blanks to NAs
tox1.m<-melt(tox1, id.vars = "Genus") 
tox1.m<-tox1.m[!is.na(tox1.m$value), ] #remove NAs 

tox2<- 
  tox1.m %>% dplyr::group_by(Genus) %>% 
  dplyr::summarise(toxins = toString(variable)) #combine toxins for each genera
tox2$ana.micro<-ifelse(grepl("anatoxin", tox2$toxins, ignore.case = T) & 
                         grepl("microcyst", tox2$toxins, ignore.case = T), "ANA+MCY", 
                       ifelse(grepl("anatox", tox2$toxins, ignore.case = T), "ANA", 
                              ifelse(grepl("microcyst", tox2$toxins, ignore.case = T), "MCY", NA)))

#combine to genus--------
#16S ASV 
d.asv.gen.tox<-
  merge(tox2, d.asv.gen, by.x="Genus", by.y="taxon6",  all.y=T) 
#anaC genus
#tox4<-merge(tox2, d.asv.gen, by.x="Genus", by.y="Taxon6", all.y=T) 

#get just cyanos
d.asv.gen.tox.cy<-d.asv.gen.tox[grepl("Cyanobacteriota", d.asv.gen.tox$taxon2), ]
#get just toxin producers
d.asv.gen.tox.prod<-d.asv.gen.tox[!is.na(d.asv.gen.tox$toxins), ]
```

edit: consolidated at station (summed), then made proportional (%)
edit: add taxonomy + toxin info to ASV
taxa = columns, samples = rows

```{r file_options3, echo = T, eval = T}
a.temp.df<-d.asv
#d.asv.gen.tox #<-- all ASV
#d.asv.gen.tox.cy #<-- just cyano-ASV
#d.asv #<-- anaC

tax.tox.asv<-a.temp.df[, c(1:4, (ncol(a.temp.df)-1), ncol(a.temp.df))] 
#taxa, toxin info
d2<-a.temp.df[, 4:(ncol(a.temp.df)-2)] #ASV table
#d2<-a.temp.df #for anaC only

#consolidate at station level
rownames(d2)<-d2[, 1]     #make first column into row names
d2<-d2[,-1]               #remove first column
d2<-as.data.frame(t(d2))  #samples must be rows!!!!
d2<-d2[!grepl("EB", rownames(d2)), ] #remove extraction blanks
d2$sample<-rownames(d2)
d2<-merge(met1[, 
               match(c("Sequencing.name.16Scyano", "Station.Name"), colnames(met1))],
               #match(c("Sequencing.name.anaC", "Station.Name"), colnames(met1))],
          d2, 
          by.y = "sample", by.x = "Sequencing.name.16Scyano", all.y = T)
         # by.y = "sample", by.x = "Sequencing.name.anaC", all.y = T)
d2<-d2[!is.na(d2$Station.Name), ]
d2$Station.Name<-as.character(d2$Station.Name)
d2.1<-d2
d2.1<-plyr::ddply(d2.1, "Station.Name", plyr::numcolwise(sum))
d2.1<-d2.1[!grepl("EB|R12", d2.1$Station.Name), ] #remove extraction blanks

#make proportional for each station
d3<-d2.1
d3<-d3[order(d3$Station.Name), ]
samps.sum.2<-samps.sum.2[order(samps.sum.2$Station.Name), ]
d3<-cbind(d3$Station.Name, 
          sweep(d3[,2:ncol(d3)], 1,
                samps.sum.2$sum, `/`)*100) #make proportional from ALL ASV 

#add tax+tox info
d4<-d3
rownames(d4)<-d4$`d3$Station.Name`
d4<-d4[, -1]
d4<-as.data.frame(t(d4))
d4$ASV<-rownames(d4)
d4<-merge(tax.tox.asv, d4, by="ASV", all.y = T, sort = F)
#d4<-merge(tax.ana, d4, by.y="ASV", by.x="Feature.ID", all.y = T, sort = F)

#consolidate at genus level, all samples
#colnames(d.asv.gen)[3]<-"taxon6"
d.asv.gen$taxon6<-gsub("Uncultured_.*","Uncultured", d.asv.gen$taxon6)
d.gen.all<-plyr::ddply(d.asv.gen, "taxon6", plyr::numcolwise(sum)) 
#consolidate at genus level
d.gen.all$taxon6[is.na(d.gen.all$taxon6)]<-"no_genus"
d.gen.all$taxon6<-gsub("t6_", "", d.gen.all$taxon6)
rownames(d.gen.all)<-d.gen.all$taxon6
d.gen.all<-d.gen.all[, -1]
d.gen.all<-as.data.frame(t(d.gen.all))
d.gen.all<-cbind(rownames(d.gen.all), 
                 d.gen.all[])
colnames(d.gen.all)[1]<-"sample"
d.gen.all$.toxin<-""
d.gen.all.m<-melt(d.gen.all, id.vars = "sample")
d.gen.all.m$num<-1:nrow(d.gen.all.m)
colnames(d4)
d.gen.all.m<-merge(d4[, match(c("Genus", "toxins", "ana.micro"), colnames(d4))], 
             d.gen.all.m, by.x="Genus", by.y = "variable", all.y=T, sort=F) 
#add toxin info
d.gen.all.m<-d.gen.all.m[!duplicated(d.gen.all.m$num), ]

#consolidate at genus level, stations consolidated
d.gen<-plyr::ddply(d4, "Genus", plyr::numcolwise(sum)) #consolidate at genus level
#colSums(d.gen[2:ncol(d.gen)]) 
#almost 100%, previously remove some ASV - this is okay
d.gen$num<-1:nrow(d.gen)
d.gen<-merge(d4[, match(c("Genus", "toxins", "ana.micro"), colnames(d4))], 
             d.gen, by = "Genus", all.y=T, sort=F) #add toxin info
d.gen<-d.gen[!duplicated(d.gen$num), ]
d.gen1<-d.gen[!is.na(d.gen$toxins), ]              #only toxin producers
d.gen1.m<-melt(d.gen1)
d.gen1.m$samp.num<-as.numeric(d.gen1.m$variable)
d.gen1.m$samp.genus<-paste(d.gen1.m$variable, d.gen1.m$Genus)

```

### environmental

```{r environmental_viz, echo=T, eval=T, cache=T}
met.tox<-met[, match(c("Extraction.name",  "Anatoxin_ELISA", "Microcytins_ELISA", 
                       "Cylindrospermopsin_ELISA", "Saxitoxin_ELISA"), colnames(met))]
colnames(met.tox)<-gsub("_ELISA", "", colnames(met.tox))
met.tox.m<-melt(met.tox, id.vars = "Extraction.name")
met.tox.m<-merge(met.tox.m, met[, match(c("Extraction.name", "SCCWRP.name", "Station.Name", 
                                        "StreamName", "RiverName", "RiverName2"), 
                                colnames(met))], 
                 by="Extraction.name", all.x = T, sort = F)
met.tox.m<-met.tox.m[!grepl("EB|R12", met.tox.m$SCCWRP.name), ]
met.tox.m<-met.tox.m[!is.na(met.tox.m$RiverName), ]
met.tox.m<-met.tox.m[order(met.tox.m$RiverName2), ]
met.tox.m$SCCWRP.name<-factor(met.tox.m$SCCWRP.name, levels = unique(met.tox.m$SCCWRP.name))
met.tox.m$value.2<-ifelse(is.na(met.tox.m$value), "absent",
                          ifelse(met.tox.m$value=="NOT ANALYZED", "no data",
                                 ifelse(met.tox.m$value=="ND", "absent","present")))
met.tox.m$value.2<-as.character(met.tox.m$value.2)

gh<-ggplot(met.tox.m)+
  geom_tile(aes(x=variable, y=SCCWRP.name, fill=value.2))+
  scale_fill_manual(values=c("#2e669e", "grey", "#f98686"))+
  facet_grid(vars(RiverName), scales = "free_y", space = "free_y")+
  labs(x="", y="Station name + sample #", fill="", 
       title="R1 2021-2022 cyanotoxins")+
  theme(plot.title = element_text(hjust=0.5), 
        axis.text.y = element_text(size=6), 
        legend.position = "top")
gh
#ggsave(plot=gh, 
#       filename = "R1_statewide_FHAB_toxin.png", 
#       path = path, 
#       height = 10, width = 8)

```

```{r toxin_v_toxinProd, cache=T, echo = T, eval=T}
#d.gen.all <-- consolidate at genus, all samples, non-prop
d.gen.all.sp<-cbind(d.gen.all$sample, 
                    d.gen.all[, colnames(d.gen.all) %in% d.asv.gen.tox.prod$Genus])
#only keep toxin producers: 25 genera
colnames(d.gen.all.sp)[1]<-"sample"
met.sp<-met1[, match(c("Sequencing.name.16Scyano", "Station.Name", "StreamName",
                       "RiverName2", "Anatoxin_ELISA", "Microcytins_ELISA", 
                       "Saxitoxin_ELISA"), colnames(met1))]
d.gen.all.sp<-d.gen.all.sp[!grepl("EB", d.gen.all.sp$sample), ] #remove EB: 89 samples
met.sp<-met.sp[met.sp$Sequencing.name.16Scyano %in% d.gen.all.sp$sample, ] #match samples

#plot genera against toxin-------
d.gen.all.sp.m<-melt(d.gen.all.sp)
d.gen.all.sp.m<-merge(d.gen.all.sp.m, met.sp, by.x="sample", 
                      by.y="Sequencing.name.16Scyano", all.x=T, sort=F)
d.gen.all.sp.m$Anatoxin_ELISA<-as.numeric(d.gen.all.sp.m$Anatoxin_ELISA)
d.gen.all.sp.m$Microcytins_ELISA<-as.numeric(d.gen.all.sp.m$Microcytins_ELISA)
d.gen.all.sp.m$Saxitoxin_ELISA<-as.numeric(d.gen.all.sp.m$Saxitoxin_ELISA)

scg1<-ggplot(d.gen.all.sp.m)+
  geom_point(aes(x=Anatoxin_ELISA, y=value))+
  facet_wrap(~variable, ncol=6)+
  labs(y="sequence abundance of toxin producer", 
       title = "R1 rivers collected 2019- 16Scyano (Monchamp)")

scg2<-ggplot(d.gen.all.sp.m)+
  geom_point(aes(x=Microcytins_ELISA, y=value))+
  facet_wrap(~variable, ncol=6)+
  labs(y="sequence abundance of toxin producer", 
       title = "R1 rivers collected 2019- 16Scyano (Monchamp)")

scg3<-ggplot(d.gen.all.sp.m)+
  geom_point(aes(x=Saxitoxin_ELISA, y=value))+
  facet_wrap(~variable, ncol=6)+
  labs(y="sequence abundance of toxin producer", 
       title = "R1 rivers collected 2019- 16Scyano (Monchamp)")

gridExtra::grid.arrange(scg1, scg2, scg3)

#ggsave(plot = scg3, 
#       filename = "R1_16Scyano_022420_75_CyanoSeq_95-97_gencol_toxinvToxinProd_sax.png",
#       path = "/Users/kylielanglois/OneDrive - SCCWRP/algae/FHABs/",
#       height = 6, width = 8)

sp.tax<-c("Calothrix", "Cylindrospermum", "Dolichospermum", "Microcoleus", 
          "Nostoc", "Tychonema")

```

```{r toxin_v_toxinProd_2, echo=T, eval=T, cache=T}
#d.gen.all.sp
met.sp$Anatoxin_ELISA<-as.numeric(met.sp$Anatoxin_ELISA)
met.sp$Saxitoxin_ELISA<-as.numeric(met.sp$Saxitoxin_ELISA)
met.sp$Microcytins_ELISA<-as.numeric(met.sp$Microcytins_ELISA)
met.eel<-met.sp[grepl("Eel", met.sp$RiverName2), ]
d.gen.all.sp.eel<-d.gen.all.sp[d.gen.all.sp$sample %in% met.eel$Sequencing.name.16Scyano, ]

df<-data.frame(genus="", 
               p.val.an="", 
               spearman.stat.an="", 
               p.val.sx="", 
               spearman.stat.sx="", 
               p.val.my="", 
               spearman.stat.my="")
dat.spear<-d.gen.all.sp.eel
met.spear<-met.eel
for (i in 2:ncol(dat.spear)) {
  c.an<-cor.test(dat.spear[, i], met.spear$Anatoxin_ELISA, method="spearman")
  c.sx<-cor.test(dat.spear[, i], met.spear$Saxitoxin_ELISA, method="spearman")
  c.my<-cor.test(dat.spear[, i], met.spear$Microcytins_ELISA, method="spearman")
  name<-colnames(dat.spear)[i]
  p.val.an<-round(c.an$p.value, 3)
  s.val.an<-round(c.an$statistic, 3)
  p.val.sx<-round(c.sx$p.value, 3)
  s.val.sx<-round(c.sx$statistic, 3)
  p.val.my<-round(c.my$p.value, 3)
  s.val.my<-round(c.my$statistic, 3)
  temp<-data.frame(genus=name,
                   p.val.an=p.val.an, 
                   spearman.stat.an=s.val.an, 
                   p.val.sx=p.val.sx, 
                   spearman.stat.sx=s.val.sx, 
                   p.val.my=p.val.my, 
                   spearman.stat.my=s.val.my)  
  df<-rbind(df, temp)
}
#write.csv(df, row.names = F, 
#          "/Users/kylielanglois/OneDrive - SCCWRP/algae/FHABs/R1_16Scyano_022420_75_CyanoSeq_95-97_gencol_toxinvToxinProd_spear_eel.csv")

```


### alpha diversity

shannon diversity index
taxa = columns, samples = rows
```{r shannon_1, echo = T}
#calculate shannon index for each sample
sh.df<-data.frame(Sample = d2$Station.Name,
              Shannon =  diversity(
                x = d2[, 3:ncol(d2)], # exclude Treatment col
                index = 'shannon'))
sh.df$num<-1:nrow(sh.df)

#16S
df.met<-merge(sh.df, met1, 
              by.x="Sample", by.y="Station.Name", all.x=T)
df.met<-df.met[!duplicated(df.met$num), ]
#anaC
#df.met<-merge(sh.df, met1, 
#by.x="Sample", by.y="Sequencing.name.anaC", all.x=T)       
```

```{r shannon_plot, echo=T, eval=T}
df.met<-df.met[!grepl("0", df.met$RiverName), ]
#plot shannon---------
col.riv<-c("#C52E19", "#FBA72A", 
           #"#54D8B1",
           "#0A9F9D",
           #"#5785C1",
           "#9986A5", 
           "#175149")
           #"#6C8645")
s.g<-ggplot(df.met, 
            aes(x=StreamName, y=Shannon, col=RiverName2))+
  geom_boxplot(fill=NA, outlier.shape = NA)+
  geom_point(aes(col=RiverName2), position = position_jitter(), 
             size=3, alpha=0.75)+
  scale_color_manual(values=col.riv)+
  theme(axis.text.x = element_text(angle = 45, hjust = 1,  size = 8))
s.g

#ggsave(plot=s.g,
#       filename = "R1_16Scyano_022420_75_cyanoOnly_stationcol_shannon.png",
#       path = "/Users/kylielanglois/OneDrive - SCCWRP/algae/FHABs/",
#       height = 6, width = 10)
```

### beta diversity

non-metric multidimensional scaling (NMDS)
taxa = columns, samples = rows
```{r nmds_1, echo=T, eval=T, cache=T}
d.nmds<-d2[ , -match("Station.Name", colnames(d2))]      #must have no character columns
rownames(d.nmds)<-d.nmds$Sequencing.name.16Scyano
d.nmds<-d.nmds[, -1]

n<-metaMDS(d.nmds, trymax = 100, trace=F)          
```

```{r nmds_2, echo=T, eval=T, cache=T}
coords<-as.data.frame(n$points)
coords2<-coords
coords2$samp<-rownames(coords2)
coords3<-merge(coords2, met1, by.x="samp", 
               #by.y="Station.Name",
               by.y = "Sequencing.name.16Scyano", 
               #by.y = "Sequencing.name.anaC", 
               all.x = T, sort = F) #merge coords with metdata for plotting
col.riv<-c("#C52E19", "#FBA72A", 
           #"#54D8B1",
           "#0A9F9D",
           #"#5785C1",
           "#9986A5", 
           "#175149")
           #"#6C8645")
coords3<-coords3[!grepl("0", coords3$StreamName), ]

pg<-ggplot()+
  #geom_segment(data=env.arrows2, 
  #             aes(xend = NMDS1, yend = NMDS2, x=0, y=0), 
  #             arrow = arrow(type = "closed", length=unit(0.1,"cm")), alpha=0.5)+
  geom_point(data=coords3, aes(x=MDS1, y=MDS2, col=RiverName2, shape=StreamName), 
             size=5, alpha=0.75, stroke=1)+
  scale_shape_manual(values=c(16, 18, 5, 1, 10, 13, 17, 15, 12))+
  scale_color_manual(values=col.riv)+
  #geom_text(data=env.arrows2, 
  #             aes(x = NMDS1, y = NMDS2, label=env))+
  ggtitle("R1 rivers collected 2019\n16Scyano (Monchamp), ASV-cyano only")
#  ggtitle("R1 rivers collected 2019\nanaC genus")
pg
#ggsave(plot=pg, 
#       filename = "R1_16Scyano_022420_75_cyanoOnly_stationcol_NMDS_riv.png",
#       path = "/Users/kylielanglois/OneDrive - SCCWRP/algae/FHABs/",
#       height = 6, width = 8)
```

```{r anosim_1, echo=T, eval=T}
d.asv.an<-d.asv
rownames(d.asv.an)<-d.asv.an$ASV
d.asv.an<-d.asv.an[, -1]
d.asv.an<-as.data.frame(t(d.asv.an))

met2<-met1[met1$Sequencing.name.16Scyano %in% rownames(d.asv.an), ]        #remove extraction blanks
#met2<-met1[met1$Sequencing.name.anaC %in% rownames(d2), ]             #remove extraction blanks

#only some columns
met2<-met2[, match(c("Sequencing.name.16Scyano", "StreamName", "RiverName", "RiverName2", 
                     "DO", "PH", "Salinity", "Temp", "TotalALK", "TotalNH3" , 
                     "DOC", "TotalHARD", "DissNO3", "DissTN", "TotalTN" ,"DissOPO4",
                     "TotalOPO4", "TotalTP", "VolumeFiltered_mL",
                     "Microscopy_Order", "Visual_Identification", "Expected",                    
                     "Anatoxin_ELISA", "Microcytins_ELISA", "Cylindrospermopsin_ELISA",    
                     "Saxitoxin_ELISA",  "anaCseq_Cyanobacteria", "anaCseq_Oscillatoria",        
                     "anaCseq_Phormidium", "anaCseq_Unassigned"), 
                   colnames(met2))]
rownames(met2)<-met2$Sequencing.name.16Scyano
met2<-met2[, -1]
```

analysis of similarity (anosim) 
```{r anosim_2, eval=F, echo=T, cache=F}
met2[] <- lapply(met2, function(x) gsub("ND", 0, x))
met2[] <- lapply(met2, function(x) gsub("NOT ANALYZED", 0, x))
met2[is.na(met2)]<-0                            # change all NAs to 0
met2<-met2[, -c(24)]                         #remove column with only 0s

df<-data.frame(variable="", 
               p.val="", 
               r.stat="")
for (i in 1:ncol(met2)) {
  a<-anosim(d.asv.an, grouping = met2[, i])     ASV = columns, samples = rows
  name<-colnames(met2)[i]
  p.val<-a$signif
  r.stat<-round(a$statistic, 3)  
  temp<-data.frame(variable=name, 
                   p.val=p.val, 
                   r.stat=r.stat)
  df<-rbind(df, temp)
}   
#get error there are no replicates within groups for anaC seq data-fine to ignore
```

```{r anosim_3, eval=F, echo=F, cache=F, warning=F}
df$p.val<-as.numeric(df$p.val)
df$r.stat<-as.numeric(df$r.stat)
df.p<-subset(df, df$p.val<1)

#write.csv(df, row.names = F, 
#          file.path(path, "R1_16Scyano_022420_75_ASV_anosim.csv"))

#kable(df.p[2:nrow(df.p), ], 
#      align="lcc", digits=2)
```

### taxonomy

taxonomy info

```{r tax_1, echo=T,  eval = T}
#use station consolidated, NOT genus consolidated
d4$final<-
  ifelse(d4$taxon2=="t2_Cyanobacteriota" & !is.na(d4$toxins), 
         "cyano-tox", 
         ifelse(d4$taxon2=="t_2Cyanobacteriota" & is.na(d4$toxins), 
                "cyano", 
                "other"))
d.fin<-d4[, c(ncol(d4), 5:(ncol(d4)-1))]
colnames(d.fin)[1]<-"final"
d.fin$final<-make.unique(d.fin$final)

#r.df<-data.frame(asv=d.asv$ASV, sums=rowSums(d[2:ncol(d.asv)]))
#summary(r.df)
```

```{r plot_tax_1, echo=T, eval=T}
d.m<-melt(d.fin)
d.m$category<-sub("\\..*", "", d.m$final)
#d.m$sample<-sub("_.*", "", d.m$variable)
#d.m$sample1<-paste(sub(".*\\.", "", d.m$sample), 
#               sub("\\..*", "", d.m$sample), sep = ".")
d.m$num<-1:nrow(d.m)
d.m<-merge(d.m, met1, by.x = "variable", by.y = "Station.Name", all.x = T)
d.m<-d.m[!duplicated(d.m$num), ]
d.m<-d.m[!grepl("R12003055", d.m$variable), ]
d.m<-d.m[, !grepl("\\.y", colnames(d.m))]
colnames(d.m)<-gsub("\\.x", "", colnames(d.m))
d.m$RiverName2[grepl("114EF", d.m$SCCWRP.name)]<-"Russian River"

g.m<-ggplot(d.m)+
  geom_bar(aes(x=variable, y=value, fill=category), 
           position = position_stack(), stat = "identity")+
  scale_fill_manual(values=c("#1f78b4", "#b2df8a", "grey"), 
                    labels=c("non-toxic cyano", "toxin producers", "other"), 
                    name="")+
  labs(x="sample", y="relative sequence abundance per station (%)", 
       title = "16Scyano sequencing data 2/24/20\nCyanoSeq BLAST taxonomy")+
  facet_grid(~RiverName2, scales = "free_x", space ="free")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 8),
        plot.title = element_text(hjust=0.5), 
        legend.position = "bottom")
g.m
#ggsave(plot = g.m, 
#       filename = "R1_16Scyano_022420_75_CyanoSeq_95-97_stationcol_prop_toxin_bar.png", 
#       path = path, 
#       height = 8, width = 14)


```


```{r plot_tax_2, echo=T, eval=T, cache=T}
#change some columns of metadata for plotting-------
met1$Date.collected<-as.Date(met1$Date.collected, format = "%m/%d/%y")
met1$plot.name<-paste(met1$Station.Name, format(as.Date(met1$Date.collected, format = "%m/%d/%y"), "%m-%d"))
met1$plot.name<-make.unique(met1$plot.name)

#plot tax--------
d1.m<-melt(d4[,c(2:4, 5:ncol(d4))])
d1.m$num<-1:nrow(d1.m)

d1.m.m<-merge(d1.m, met1, 
              by.x = "variable", by.y = "Station.Name",
              #by.x = "variable", by.y = "Sequencing.name.anaC",
              all.x = T)
d1.m.m<-d1.m.m[!duplicated(d1.m.m$num), ]
d1.m.m<-d1.m.m[!grepl("0", d1.m.m$StreamName), ]
d1.m.m<-d1.m.m[!is.na(d1.m.m$ana.micro), ]

d1.m.m.1<-d1.m.m
d1.m.m.1$variable<-as.character(d1.m.m.1$variable)
d1.m.m.1$group<-paste(d1.m.m.1$variable, d1.m.m.1$Genus)
d1.m.m.summ<-aggregate(d1.m.m.1$value, 
                       by=list(total=d1.m.m.1$group), FUN=sum)
d1.m.m.summ$sample<-sub(" .*", "", d1.m.m.summ$total)
d1.m.m.summ$gen<-sub(".* ", "", d1.m.m.summ$total)
d1.m.m.summ$x<-round(d1.m.m.summ$x, 2)

bg<-ggplot(d1.m.m)+
  geom_bar(aes(x=plot.name, y=value, fill=ana.micro), 
           stat = "identity", position = position_stack())+
  scale_fill_manual(values=c("#9970ab", "#80E8C0", "#1b7837"), name="")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 8),
        plot.title = element_text(hjust=0.5), 
        legend.position = "bottom")+
  facet_grid(~RiverName2, scales = "free_x", space ="free")+
  labs(x="Station Code + Collection Month-Day (2019)", 
       y="proportion of sequences (%) per station", 
       title = "16Scyano sequencing data 2/24/20\nCyanoSeq BLAST taxonomy, putative toxigenic cyanobacteria")
bg
```

```{r plot_tax_3, echo=T, eval=T, cache=T}
col.cy.gen<-c("#58C890", "#801010" ,"#6C8645", "#D84040", "#F8E850", 
              "#205080", "#80E8C0", "#F88070",  "grey","#803000","#F09008",
              "#40A018","#B86830", "#184830","#D8C830",  "#4890C0" ,"#F8C0B8",
              "#98E0F0", "#D85020", "#505020", "#af8dc3", "#B0A870", "#F85048", 
              "#F89040", "#903000", "#F8D068", "#D04008",
              "#98F088", "#40A078", "#90C8D8")

ag<-ggplot(d1.m.m)+
  geom_bar(aes(x=plot.name, y=value, fill=Genus), 
           stat = "identity", position = position_stack())+
  scale_fill_manual(values=col.cy.gen)+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 8),
        plot.title = element_text(hjust=0.5), 
        legend.position = "bottom", 
        legend.text = element_text(size = 12))+
  guides(fill=guide_legend(ncol = 6))+
  facet_grid(ana.micro~RiverName2, scales = "free", space ="free")+
  labs(x="Station Code + Collection Month-Day (2019)", 
      # y="# of sequences per sample",
       y="proportion of sequences (%) per station", 
       #title = "anaC sequencing data 1/26/23\nNCBI BLAST taxonomy")+
       title = "16Scyano sequencing data 2/24/20\nCyanoSeq BLAST taxonomy, putative toxigenic cyanobacteria")
ag

#ggsave(plot = ag, 
#       filename = "R1_16Scyano_022420_75_CyanoSeq_95-97_gencol_stationcol_My_An_toxin_bar_stream.png", 
#       path = path, 
#       height = 10, width = 14)

```

```{r environmental_tax, echo=T, eval=T, cache=T}
#d.gen1.m <--station consolidated
#d.gen.all.m <--all stations
d.gen.all.m$ana.micro<-ifelse(d.gen.all.m$Genus==".toxin", 0, 
                              d.gen.all.m$ana.micro)

#only ana and micro toxin prods
d.gen2<-d.gen.all.m[!is.na(d.gen.all.m$ana.micro), ] 
d.gen2$value.2<-ifelse(d.gen2$value>50, "1. >50%", 
                ifelse(d.gen2$value>25 & d.gen2$value<49, "2. 25-50%", 
                ifelse(d.gen2$value>1 & d.gen2$value<24, "3. 1-25%", 
                ifelse(d.gen2$value>0 & d.gen2$value<1, "4. 0-1%", 
                                            "5. absent"))))
d.gen2$num<-1:nrow(d.gen2)
d.gen2<-merge(d.gen2, met[, match(c("Station.Name", "Sequencing.name.16Scyano", 
                                    "SCCWRP.name","RiverName", "RiverName2", 
                                    "StreamName", "Anatoxin_ELISA", 
                                    "Microcytins_ELISA"), 
                                   colnames(met))], 
              #by.x = "variable", by.y = "Station.Name", all.x = T, sort = F)
              by.x = "sample", by.y = "Sequencing.name.16Scyano", all.x = T, sort = F)
d.gen2<-d.gen2[order(d.gen2$RiverName2), ]
d.gen2$SCCWRP.name<-factor(d.gen2$SCCWRP.name, levels = unique(d.gen2$SCCWRP.name))

d.gen2$Anatoxin_ELISA<-ifelse(d.gen2$Anatoxin_ELISA=="ND", NA, 
                       ifelse(d.gen2$Anatoxin_ELISA=="NOT ANALYZED", NA, 
                              "ANA"))
d.gen2$Microcytins_ELISA<-ifelse(d.gen2$Microcytins_ELISA=="ND", NA, 
                          ifelse(d.gen2$Microcytins_ELISA=="NOT ANALYZED", NA, 
                                 "MCY"))
d.gen2$toxin.combo<-paste(d.gen2$Anatoxin_ELISA, d.gen2$Microcytins_ELISA)
d.gen2$toxin.combo<-ifelse(d.gen2$toxin.combo=="NA NA", "", 
                    ifelse(d.gen2$toxin.combo=="ANA NA", "ANA", 
                    ifelse(d.gen2$toxin.combo=="NA MCY", "MCY", 
                           d.gen2$toxin.combo)))

d.gen2<-d.gen2[!duplicated(d.gen2$num), ]
d.gen2<-d.gen2[!grepl("EB|EF", d.gen2$Station.Name), ]

d.gen2<-d.gen2[order(d.gen2$toxin.combo), ]
d.gen2$SCCWRP.name<-factor(d.gen2$SCCWRP.name, levels=unique(d.gen2$SCCWRP.name))

dh<-ggplot(d.gen2)+
  geom_tile(aes(x=Genus, y=SCCWRP.name, fill=value.2))+
  geom_point(aes(x=1, y=SCCWRP.name, col=toxin.combo, shape=toxin.combo), 
             size=2)+
  scale_fill_manual(values=c("#016c59", "#67a9cf", "#d0d1e6", 
                             "#f6eff7", "white"))+
  scale_color_manual(values=c(NA, "black", "black", "black"), 
                     name="toxin present")+
  scale_shape_manual(values=c(NA, 1, 8, 4), name="toxin present")+
  facet_grid(vars(RiverName), scales = "free_y", space = "free_y")+
  labs(x="", y="Station name + rep", fill="rel. abundance", 
       title="R1 2021-2022 cyanotoxin producers")+
  theme(plot.title = element_text(hjust=0.5), 
        axis.text.y = element_text(size=6), 
        axis.text.x = element_text(angle = 45, hjust = 1),
        axis.ticks.y = element_blank(),
        legend.position = "right")
dh

#ggsave(plot = dh, 
#       filename = "R1_16Scyano_022420_75_CyanoSeq_95-97_gencol_heat.png",
#       path = path, 
#       height = 12, width = 9)

```



### scope vs seq

```{r tax_microscopy_1, echo = T, eval = T, cache=T}
met3<-met1[, match(c("Station.Name", "Expected", "RiverName2"), colnames(met1))]

met4<- 
  met3 %>% group_by(Station.Name) %>% 
  summarise(Expected = toString(Expected)) #combine expected for each station
met4$Expected2<-
  sapply(strsplit(met4$Expected, ", "), 
         function(x) paste(unique(x), collapse = ", ")) #remove duplicates
met4$num<-1:nrow(met4)
#merge to get river name
met4<-merge(met4, met3[, c(1, 3)], by="Station.Name", all.x = T) 
met4<-met4[!duplicated(met4$num), ]
#remove blanks and random sample
met4<-met4[!grepl("EB|R12", met4$Station.Name), ] 

#write.csv(met4, row.names = F, 
#          "/Users/kylielanglois/OneDrive - SCCWRP/algae/FHABs/R1_statewide_FHAB_metadata_expected_summ.csv")
met5<-read.csv(
  "/Users/kylielanglois/OneDrive - SCCWRP/algae/FHABs/R1_statewide_FHAB_metadata_expected_summ.csv")
expected.df<-data.frame(combo=c(met5$combo1, met5$combo2, met5$combo3, 
                                met5$combo4, met5$combo5, met5$combo6, 
                                met5$combo7, met5$combo8, met5$combo9))
expected.df<-as.data.frame(
  expected.df[!(is.na(expected.df$combo) | expected.df$combo==""), ])
colnames(expected.df)<-"combo"

d1.m.m$combo<-paste(d1.m.m$variable, d1.m.m$Genus)
d1.m.ex<-merge(expected.df, d1.m.m, by = "combo", all.x = T, sort = F)
d1.m.ex<-d1.m.ex[!is.na(d1.m.ex$Genus), ]

eg<-ggplot(d1.m.ex)+
  geom_bar(aes(x=plot.name, y=value, fill=Genus), 
           stat = "identity", position = position_stack())+
  #scale_fill_manual(values=col.cy.gen)+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 8),
        plot.title = element_text(hjust=0.5), 
        legend.position = "bottom", 
        legend.text = element_text(size = 12))+
  guides(fill=guide_legend(ncol = 6))+
  #facet_grid(ana.micro~RiverName2, scales = "free", space ="free")+
  labs(x="Station Code + Collection Month-Day (2019)", 
      # y="# of sequences per sample",
       y="proportion of sequences (%) per station", 
       #title = "anaC sequencing data 1/26/23\nNCBI BLAST taxonomy")+
       title = "16Scyano sequencing data 2/24/20\nCyanoSeq BLAST taxonomy, 
       matched to microscopy ID")
eg

#ggsave(plot = eg, 
#       filename = "R1_16Scyano_022420_75_CyanoSeq_95-97_gencol_stationcol_bar_stream_exp.png", 
#       path = path, 
#       height = 5, width = 7)
```


```{r tax_microscopy_2, eval=T, echo=T, cache=T}
#for station consolidated--------
#d.gen1.m.scope
#d.gen.all.m <--all stations
d.gen1.m<-d.gen1.m[!grepl("num", d.gen1.m$samp.genus), ] #only toxin producers
#write.csv(d.gen1.m, row.names = F, 
#          file.path(path, #"R1_16Scyano_022420_75_toxinprod_prop_melt.csv"))
d.gen1.m<-
  read.csv(file.path(path, "R1_16Scyano_022420_75_toxinprod_prop_melt.csv"))
d.gen1.m<-subset(d.gen1.m, d.gen1.m$keep==1)

d.gen1.m.1<-merge(met3, d.gen1.m, by.x="Station.Name", 
                  by.y="variable", all.y=T) #get scope ID
#d.gen1.m.1$Expected2<-gsub(",.*", "", d.gen1.m.1$Expected2)

dgmsc2 <- summarise(group_by(d.gen1.m.1,Expected,Genus),count =n())

#for all samples--------
#d.gen.all
d.gen.all.m<-melt(d.gen.all, id.vars = "sample")
d.gen.all.m.1 <- d.gen.all.m %>%
  group_by(sample) %>%
  mutate(top_tax = variable[which.max(value)]) %>%
  ungroup()
d.gen.all.m.1<-d.gen.all.m.1[!duplicated(d.gen.all.m.1$sample), ]
d.gen.all.1<-merge(d.gen.all.m.1[, match(c("sample", "top_tax"), 
                                         colnames(d.gen.all.m.1))], 
                   met1[, match(c("Sequencing.name.16Scyano", "Expected"), 
                                colnames(met1))], 
                   by.x="sample", by.y="Sequencing.name.16Scyano", 
                   all.x=T, sort=F)

dgmsc3 <- summarise(group_by(d.gen.all.1,Expected,top_tax),count =n())
dgmsc3$top_tax<-gsub("no_genus", "\\.no_genus", dgmsc3$top_tax)
dgmsc3<-dgmsc3[!grepl("0|Sponge|Diatom", dgmsc3$Expected), ]
dgmsc3$count<-as.numeric(dgmsc3$count)
#Add column for orders
dgmsc3 <- dgmsc3 %>% 
  mutate(Micros_Order = case_when(Expected == 'Oscillatoria' ~ 'Oscillatoriales',
                         Expected == 'Phormidium' ~ 'Oscillatoriales',
                         Expected == 'Unknown' ~ 'NA',
                         TRUE ~ 'Nostocales'))
dgmsc3<-merge(dgmsc3, tax.select, by.x="top_tax", by.y="taxon6", 
              all.x=T, sort=F)
dgmsc3<-dgmsc3[rev(order(dgmsc3$Micros_Order)), ]
dgmsc3$Expected<-factor(dgmsc3$Expected, levels=unique(dgmsc3$Expected))

ord.col<-c("grey","#3288BD", "#F46D43")

scg<-ggplot(dgmsc3, 
            aes(x=Expected, y=top_tax))+
  geom_point(aes(fill = Micros_Order, size = count),
             shape = 21, color = "black", alpha = 0.75)+
  geom_text(aes(label=count), size = 3.5)+
  scale_fill_manual(values = ord.col, name="Microscopy\nOrder")+ 
  scale_size_continuous(range = c(4,15))+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 8), 
        plot.title = element_text(hjust = 0.5))+
  guides(size="none")+
  labs(x="Microscopy top ID", y="Sequencing top ID", 
       title="16Scyano sequencing data 2/24/20\nCyanoSeq BLAST taxonomy 
       matched to microscopy ID")
scg
#ggsave(plot=scg, 
#       filename = "R1_16Scyano_022420_75_CyanoSeq_95-97_gencol_dot_exp.png", 
#       path = path,
#       height=6, width=9)
```

```{r final, echo=F, eval=T}
#code is finished :)
```